### PR TITLE
new file

### PR DESCRIPTION
**Pull request by PrajwalYadav**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by Divyani286 — APPROVED** [review]

File: Team_02/society_manage/final/complaints.c [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include "soc.h"
+
+#define size 1000
+
+/*struct complaints_D
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c]: remove commented data if it's not needed... if its needed then remove comment

File: Team_02/society_manage/final/complaints.c [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+/*write the complaints data*/
+void write_complaints_D(struct comData com)
+{
+    int fd3;
+    fd3 = open(COMPLAINT_DATAFILE, O_RDWR | O_CREAT | O_APPEND, 0644);
+    if (fd3 < 0) {
+      perror("file open failed...");
+    }
+      write(fd3, (void *)&com, sizeof(struct comData));
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c]: remove extra spaces...

File: Team_02/society_manage/final/complaints.c [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+    int key3 = com_data.flat_num3 % size;
+    newNode3->next3 = NULL;
+    newNode3->prev3 = NULL;
+
+    if(arr3[key3] == NULL) {
+      newNode3->next3 = NULL;
+      newNode3->prev3 = NULL;
+      arr3[key3] = newNode3;
+    }
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c:101]: else should be next to ' { ' at same line

File: Team_02/society_manage/final/complaints.c [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    else {
+      struct complaint* temp3 = arr3[key3];
+      while(temp3->next3 != NULL){
+        temp3 = temp3->next3;
+        }
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c:106]: align closing '{' while loop braces to proper line space

File: Team_02/society_manage/final/complaints.c [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    newNode3->prev3 = NULL;
+
+    if(arr3[key3] == NULL) {
+      newNode3->next3 = NULL;
+      newNode3->prev3 = NULL;
+      arr3[key3] = newNode3;
+    }
+    else {
+      struct complaint* temp3 = arr3[key3];
+      while(temp3->next3 != NULL){
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c]: space between ' ) { '

File: Team_02/society_manage/final/complaints.c [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        temp3 = temp3->next3;
+        }
+        temp3->next3 = newNode3;
+        newNode3->prev3 = temp3;
+        //newNode->next = NULL;
+    }
+    //printf("Node inserted\n");
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c]: remove commented part

File: Team_02/society_manage/final/complaints.c [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+        ptr3 = ptr3 -> next3;
+    }
+
+    if (flag == 1) {
+      printf("\n\n\tComplaint flat_num not found\n");
+    }
+  }
+}
+
+/*COMPLAINTS DELETION*/
```
Comment (shankar7791) [inline Team_02/society_manage/final/complaints.c:211]: remove commented code

File: Team_02/society_manage/final/maintenance.c [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+  }
+  close(fd1);
+}
+
+/*society insertion*/
+void insert_maint(struct maintData maint_data)
+{
+      //create a newnode with value
+      struct maintenance *newNode1 = (struct maintenance*)malloc(sizeof(struct maintenance));
+      newNode1->md.index1 = maint_data.index1;
```
Comment (shankar7791) [inline Team_02/society_manage/final/maintenance.c]: make sure all are the 4 space or tabs are spaces

**Review by shankar7791 — COMMENTED** [review]
solve all changes....
and look once again coding styale of all project files

**Review by shankar7791 — APPROVED** [review]

**Pull request merged**